app/graphql/types/dynamic_field_group_child_type.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Types
+  class DynamicFieldGroupChildType < Types::BaseUnion
+    description "objects that can be children of a dynamic field group"
+    possible_types Types::DynamicFieldGroupType, Types::DynamicFieldType
+    def self.resolve_type(object, _context)
+      if object.is_a?(DynamicFieldGroup)
+        Types::DynamicFieldGroupType
+      else
```

Comment:
I would recommend explicitly checking that its a "DynamicField" and then throwing an error if its not.